### ISSUE TITLE
program.loc.start.column is always zero

### ISSUE THREAD

**Issue by jdmota** [babel/babel-eslint#526]
`program.loc.start.column` seems to be always zero. Is this intended? `espree` gives `5` instead of `0`.

```js
const babel = require("babel-eslint");
const espree = require("espree");

const code = `/*
foo*/function bar() {
  console.log();
}`;

console.log(babel.parse(code).loc);
console.log(espree.parse(code,{loc: true}).loc);
```

![image](https://user-images.githubusercontent.com/18088420/31299311-c4db4fd2-aae5-11e7-8db4-64f04194a0c7.png)

**Comment by JamesHenry** [comment]
Thanks @jdmota. This is a babylon thing (i.e. nothing specific to babel-eslint), I reported it here: https://github.com/babel/babylon/issues/673

**Comment by jdmota** [comment]
Thanks for looking!

Probably there is a bug in `babylon`, but `babylon` can choose how to define the start position of the program independently of other parsers right?

My issue here was that `babel-eslint` should output the same as `espree`.

Am I missing something?

**Comment by JamesHenry** [comment]
Well, my thinking was that given that it hasn't even been acknowledged as an _intentional_ difference by babylon yet, it didn't make sense to have a separate thread on babel-eslint to discuss modifying the AST to appear the same as espree's.

The current behaviour in babylon is either:

1) An unintentional bug
In which case it will be fixed there, and no changes are relevant for babel-eslint

2) An accepted/deliberate divergence
In which case the babylon issue should be closed and we can think about transforming the AST in babel-eslint

In my mind, either scenario first requires a decision to be taken about the babylon behaviour, so I wanted to route all attention to that issue.

As you can see, though, based on my research on that issue, it is not a cut and dry situation - some parsers do one thing, some do another - so I imagine that is one of the reasons it's not been discussed any further 😅 

Happy to reopen the issue for visibility in the meantime

**Comment by jdmota** [comment]
Thanks for reopening.

I think I forgot to make something clear. The behaviour of `babel-eslint` is not just different but wrong, because it's like the Program node starts in the middle of a comment.

`babylon` and `espree` are different, but I would not say that they are wrong.

```
babel-eslint  SourceLocation {
  start: Position { line: 2, column: 0 },
  end: Position { line: 4, column: 1 } }

babylon  SourceLocation {
  start: Position { line: 1, column: 0 },
  end: Position { line: 4, column: 1 } }

espree  SourceLocation {
  start: Position { line: 2, column: 5 },
  end: Position { line: 4, column: 1 } }
```

**Comment by JamesHenry** [comment]
Ah ok, sorry, I did not get that all three are different!

**Comment by jdmota** [comment]
https://github.com/babel/babel-eslint/blob/5742b713746accacfcf4dfae1cb67bcefcc6d43a/babylon-to-espree/attachComments.js#L20-L23

If I uncomment these lines, I think the problem is fixed. Should I send a PR?

**Comment by JamesHenry** [comment]
I'll cheekily defer to @hzoo, who comes up via a "git blame" 😁

**Comment by jdmota** [comment]
I just found another difference between `babel-eslint` and `espree` related with the location of the Program node and comments.

```js

/*
foo
*/

```

```
babel-eslint  SourceLocation {
  start: Position { line: 4, column: 2 },
  end: Position { line: 5, column: 0 } }
babylon  SourceLocation {
  start: Position { line: 1, column: 0 },
  end: Position { line: 5, column: 0 } }
espree  SourceLocation {
  start: Position { line: 1, column: 0 },
  end: Position { line: 5, column: 0 } }
```

And in this case, `babel-eslint` does not output the `trailingComments`.

```
babel-eslint  undefined
espree  [ { type: 'Block',
    value: '\nfoo\n',
    start: 1,
    end: 10,
    range: [ 1, 10 ],
    loc: { start: [Object], end: [Object] } } ]
```

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
